sync.go

```diff
@@ -16,11 +16,34 @@ import (
 )
 
 type syncData struct {
-	mu                    sync.Mutex
-	rpcClient             *chain.RPCClient
-	cancelSync            context.CancelFunc
-	syncProgressListeners []SyncProgressListener
-	rescanning            bool
+	mu                             sync.Mutex
+	rpcClient                      *chain.RPCClient
+	cancelSync                     context.CancelFunc
+	syncProgressListeners          []SyncProgressListener
+	estimatedSyncProgressListeners []EstimatedSyncProgressListener
+	rescanning                     bool
+	syncing                        bool
```

Comment:
Can move these new properties (L25-46) to an `activeSyncData` struct and then embed the new struct here.